marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs

```diff
@@ -258,6 +258,7 @@ class Flow {
     this.globalContext.sourceDatabase = combinedOptions.sourceDatabase || this.globalContext.sourceDatabase;
 
     if (!(combinedOptions.noBatchWrite || combinedOptions.disableJobOutput)) {
+      Object.assign(flowStep.options, combinedOptions);
```

Comment:
I think that because flowStep is used later on, modifying it with combinedOptions could lead to unexpected results. 

How about this instead:

```
const flowStepWithOptions = Object.assign({}, flowStep.options, combinedOptions);
```

And then pass in flowStepWithOptions instead of flowStep.

I can't identify a bug this is preventing, but I think it's safer not to modify flowStep since it's used for a lot of other stuff.

Reply:
I suppose a step could be used twice in the same flow, and altering it could mess things up. I’ll change this.